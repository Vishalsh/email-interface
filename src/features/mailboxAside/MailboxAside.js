import React from 'react';
import { Link } from 'react-router-dom';

import Button from 'components/button/Button';
import Folders from 'features/mailboxAside/folders/Folders';
import Categories from 'features/mailboxAside/categories/Categories';
import Labels from 'features/mailboxAside/labels/Labels';
import routes from 'constants/routes';
import classes from './MailboxAside.module.scss';

const MailboxAside = () => {
  return (
    <section className={classes.mailboxAside}>
      <Button type="primary" onClick={() => alert('compose')}>
        Compose Mail
      </Button>

      <section>
        <h4 className={classes.heading}>FOLDERS</h4>
        <Folders />
      </section>

      <section>
        <h4 className={classes.heading}>CATEGORIES</h4>
        <Categories />
      </section>

      <section>
        <h4 className={classes.heading}>LABELS</h4>
        <Labels />
      </section>
    </section>
  )
};

export default MailboxAside;
