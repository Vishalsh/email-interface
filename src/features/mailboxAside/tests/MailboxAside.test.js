import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { MailboxAside } from '../MailboxAside';

describe('MailboxAside', () => {
  const props = {
    toggleCreateEmailPopup: jest.fn()
  };

  const component = shallow(<MailboxAside {...props}/>);

  it('should render the component', () => {
    expect(toJson(component)).toMatchSnapshot();
  });
});