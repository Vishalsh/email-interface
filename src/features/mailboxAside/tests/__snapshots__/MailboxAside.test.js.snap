// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MailboxAside should render the component 1`] = `
<section
  className="mailboxAside"
>
  <Button
    className=""
    disabled={false}
    onClick={[Function]}
    type="primary"
  >
    Compose Mail
  </Button>
  <section>
    <h4
      className="heading"
    >
      FOLDERS
    </h4>
    <Folders />
  </section>
  <section>
    <h4
      className="heading"
    >
      CATEGORIES
    </h4>
    <Categories />
  </section>
  <section>
    <h4
      className="heading"
    >
      LABELS
    </h4>
    <Labels />
  </section>
</section>
`;
